Moonscraper Chart Editor/Assets/Scripts/Game/Charts/IO/Midi/MidReader.cs

```diff
@@ -445,104 +445,105 @@ private static void ReadNotes(IList<MidiEvent> track, Song song, Song.Instrument
-                            var b2 = se2.GetData();
-                            if (b2.Length == 8 && b2[5] == bytes[5] && b2[7] == 0)
+                            var se2 = tapAndOpenEvents[j] as SysexEvent;
+                            if (se2 != null)
                             {
-                                endPos = (uint)(se2.AbsoluteTime - tick);
+                                var b2 = se2.GetData();
+                                if (b2.Length == 8 && b2[5] == bytes[5] && b2[7] == 0)
+                                {
+                                    endPos = (uint)(se2.AbsoluteTime - tick);
 
-                                if (endPos > 0)
-                                    --endPos;
+                                    if (endPos > 0)
+                                        --endPos;
 
-                                break;
+                                    break;
+                                }
                             }
                         }
-                    }
 
-                    int index, length;
-                    SongObjectCache<Note> notes;
-                    if (instrument == Song.Instrument.Unrecognised)
-                        notes = unrecognised.notes;
-                    else
-                        notes = song.GetChart(instrument, difficulty).notes;
-                    SongObjectHelper.GetRange(notes, tick, tick + endPos, out index, out length);
-                    for (int k = index; k < index + length; ++k)
-                    {
-                        switch (gameMode)
+                        int index, length;
+                        SongObjectCache<Note> notes = song.GetChart(instrument, difficulty).notes;
+                        SongObjectHelper.GetRange(notes, tick, tick + endPos, out index, out length);
+                        for (int k = index; k < index + length; ++k)
                         {
-                            case Chart.GameMode.Guitar:
-                                notes[k].guitarFret = Note.GuitarFret.Open;
-                                break;
-                            // Usually not used, but in the case that it is, it should work properly
-                            case Chart.GameMode.GHLGuitar:
-                                notes[k].ghliveGuitarFret = Note.GHLiveGuitarFret.Open;
-                                break;
+                            switch (gameMode)
+                            {
+                                case Chart.GameMode.Guitar:
+                                    notes[k].guitarFret = Note.GuitarFret.Open;
+                                    break;
+                                // Usually not used, but in the case that it is, it should work properly
+                                case Chart.GameMode.GHLGuitar:
+                                    notes[k].ghliveGuitarFret = Note.GHLiveGuitarFret.Open;
+                                    break;
+                            }
```

Comment:
Add default case and Debug.Assert false with a message.

Reply:
Done 👍